src/libraries/SqrtPriceMath.sol

```diff
@@ -238,11 +238,8 @@ library SqrtPriceMath {
     {
         uint256 numerator = absDiff(sqrtPriceAX96, sqrtPriceBX96);
         uint256 denominator = FixedPoint96.Q96;
-        uint256 _liquidity;
-        assembly ("memory-safe") {
-            // avoid implicit upcasting
-            _liquidity := and(liquidity, 0xffffffffffffffffffffffffffffffff)
-        }
+        uint256 _liquidity = uint256(liquidity);
```

Comment:
hm but why do we have a comment that says "avoid implicit upcasting".. feels like this is what we/someone was trying to avoid?

Reply:
it was just done in PR #676 to save gas. But now that we've added masking and optimiser runs are higher, it doesnt save gas anymore.